src/solver/conv_ck_igemm_fwd_bias_activ.cpp

```diff
@@ -0,0 +1,421 @@
+    case miopenFloat:
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+#endif
+}
+
+bool PerformanceConfigConvCKIgemmFwdBiasActiv::SetNextValue(const FusionContext& ctx)
+{
+    if(this->total_size == -1)
+    {
+        this->HeuristicInit(ctx);
+        assert(this->total_size != -1);
+        return true;
+    }
+    if((this->index + 1) < this->total_size)
+    {
+        ++this->index;
+        const auto conv = GetInstances();
+        this->kernel_id = conv[this->index]->GetTypeString();
+        return true;
+    }
+    return false;
+}
+
+bool PerformanceConfigConvCKIgemmFwdBiasActiv::IsValidValue() const
+{
+    return this->index < this->total_size;
+}
+
+bool PerformanceConfigConvCKIgemmFwdBiasActiv::IsValid(const FusionContext& ctx) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    return false;
+#else
+    // Extract convolution problem from the fusion context.
+    const auto& problem = ctx.problem.GetConvProblem(0, conv::Direction::Forward);
+    switch(problem.conv_problem.GetInDataType())
+    {
+    case miopenInt8:
+    case miopenHalf: return CheckIsSupportCKArgs<ck::half_t>(problem);
+    case miopenFloat:
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+    return false;
+#endif
+}
+
+bool PerformanceConfigConvCKIgemmFwdBiasActiv::operator==(
+    const PerformanceConfigConvCKIgemmFwdBiasActiv& other) const
+{
+    return this->index == other.index && this->total_size == other.total_size;
+}
+
+PerformanceConfigConvCKIgemmFwdBiasActiv
+ConvCKIgemmFwdBiasActiv::GetDefaultPerformanceConfig(const FusionContext& ctx) const
+{
+    PerformanceConfigConvCKIgemmFwdBiasActiv pp;
+    pp.HeuristicInit(ctx);
+    return pp;
+}
+
+bool ConvCKIgemmFwdBiasActiv::IsValidPerformanceConfig(
+    const FusionContext& ctx, const PerformanceConfigConvCKIgemmFwdBiasActiv& config) const
+{
+    return config.IsValid(ctx);
+}
+
+PerformanceConfigConvCKIgemmFwdBiasActiv
+ConvCKIgemmFwdBiasActiv::Search(const FusionContext& ctx, const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, invoke_ctx);
+}
+
+bool ConvCKIgemmFwdBiasActiv::IsApplicable(const FusionContext& ctx) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    return false;
+#else
+    const auto& problem = ctx.problem.GetConvProblem(0, conv::Direction::Forward);
+
+    if(problem.conv_problem.GetConv().attribute.deterministic)
+        return false;
+
+    if(problem.conv_problem.GetInDataType() != problem.conv_problem.GetWeightsDataType() ||
+       problem.conv_problem.GetWeightsDataType() != problem.conv_problem.GetOutDataType() ||
+       problem.conv_problem.GetInDataType() != problem.conv_problem.GetOutDataType())
+        return false;
+
+    if(!problem.direction.IsForward())
+        return false;
```

Comment:
Is it possible provided that `problem = ctx.problem.GetConvProblem(0, conv::Direction::Forward)`?

Reply:
Agreed! so I removed that line.